wgpu-hal/src/dx12/view.rs

```diff
@@ -67,67 +69,81 @@ impl ViewDescriptor {
                 }
             }
             wgt::TextureViewDimension::D2 | wgt::TextureViewDimension::D2Array
                 if self.multisampled =>
             {
                 desc.ViewDimension = d3d12::D3D12_SRV_DIMENSION_TEXTURE2DMSARRAY;
-                *desc.u.Texture2DMSArray_mut() = d3d12::D3D12_TEX2DMS_ARRAY_SRV {
-                    FirstArraySlice: self.array_layer_base,
-                    ArraySize: self.array_layer_count,
+                unsafe {
+                    *desc.u.Texture2DMSArray_mut() = d3d12::D3D12_TEX2DMS_ARRAY_SRV {
+                        FirstArraySlice: self.array_layer_base,
+                        ArraySize: self.array_layer_count,
+                    }
                 }
             }
             wgt::TextureViewDimension::D2 | wgt::TextureViewDimension::D2Array => {
                 desc.ViewDimension = d3d12::D3D12_SRV_DIMENSION_TEXTURE2DARRAY;
-                *desc.u.Texture2DArray_mut() = d3d12::D3D12_TEX2D_ARRAY_SRV {
-                    MostDetailedMip: self.mip_level_base,
-                    MipLevels: self.mip_level_count,
-                    FirstArraySlice: self.array_layer_base,
-                    ArraySize: self.array_layer_count,
-                    PlaneSlice: 0,
-                    ResourceMinLODClamp: 0.0,
+                unsafe {
+                    *desc.u.Texture2DArray_mut() = d3d12::D3D12_TEX2D_ARRAY_SRV {
+                        MostDetailedMip: self.mip_level_base,
+                        MipLevels: self.mip_level_count,
+                        FirstArraySlice: self.array_layer_base,
+                        ArraySize: self.array_layer_count,
+                        PlaneSlice: 0,
+                        ResourceMinLODClamp: 0.0,
+                    }
                 }
             }
             wgt::TextureViewDimension::D3 => {
                 desc.ViewDimension = d3d12::D3D12_SRV_DIMENSION_TEXTURE3D;
-                *desc.u.Texture3D_mut() = d3d12::D3D12_TEX3D_SRV {
-                    MostDetailedMip: self.mip_level_base,
-                    MipLevels: self.mip_level_count,
-                    ResourceMinLODClamp: 0.0,
+                unsafe {
+                    *desc.u.Texture3D_mut() = d3d12::D3D12_TEX3D_SRV {
+                        MostDetailedMip: self.mip_level_base,
+                        MipLevels: self.mip_level_count,
+                        ResourceMinLODClamp: 0.0,
+                    }
                 }
             }
             wgt::TextureViewDimension::Cube if self.array_layer_base == 0 => {
                 desc.ViewDimension = d3d12::D3D12_SRV_DIMENSION_TEXTURECUBE;
-                *desc.u.TextureCube_mut() = d3d12::D3D12_TEXCUBE_SRV {
-                    MostDetailedMip: self.mip_level_base,
-                    MipLevels: self.mip_level_count,
-                    ResourceMinLODClamp: 0.0,
+                unsafe {
+                    *desc.u.TextureCube_mut() = d3d12::D3D12_TEXCUBE_SRV {
```

Comment:
This has scary enough ramifications for doing wrong, that explicit unsafe is fine here.